src/components/Modals/PrintModal.vue

```diff
@@ -0,0 +1,288 @@
+<script setup>
+import { ref, watch } from 'vue'
+import { Dialog, DialogPanel, DialogTitle, TransitionChild, TransitionRoot } from '@headlessui/vue'
+import { MinusSmallIcon } from '@heroicons/vue/20/solid'
+import { PrinterIcon } from '@heroicons/vue/24/outline'
+import { ExclamationCircleIcon } from '@heroicons/vue/24/outline'
+
+const props = defineProps({
+  showNotification: Boolean,
+  validQRCode: Boolean
+})
+
+const notificationContent = ref({
+  titleText: '',
+  messageText: '',
+  buttonText: ''
+})
+if (props.validQRCode) {
+  notificationContent.value.titleText = 'Select items to print'
+  notificationContent.value.messageText = ''
+  notificationContent.value.buttonText = 'Print'
+} else {
+  notificationContent.value.titleText = 'Error!'
+  notificationContent.value.messageText = 'Please scan a valid QR code.'
+  notificationContent.value.buttonText = 'Try Again'
+}
+const disableButton = ref(false)
+const open = ref(false)
+const emit = defineEmits(['update:show-modal', 'fire-function'])
+const updateLocalShowNotification = (value) => {
+  open.value = value
+  emit('update:show-modal', value)
+}
+
+const fireFunction = () => {
+  if (props.validQRCode) {
+    emit('fire-function')
+    notificationContent.value.titleText = 'Printing...'
+    notificationContent.value.messageText = 'Please wait while we print your pass.'
+    disableButton.value = true
+    console.log('printing...')
+
+    setTimeout(() => {
+      window.location.reload()
+    }, 3000)
+  } else {
+    window.location.reload()
+    console.log('refreshing page...')
+  }
+}
+
+const printOptions = [
+  {
+    id: 'name',
+    name: 'Name',
+    label: 'Name',
+    description: 'Print your full name'
+  },
+  {
+    id: 'email',
+    name: 'Email',
+    label: 'Email',
+    description: 'Print your email'
+  },
+  {
+    id: 'org',
+    name: 'Orgnaisation',
+    label: 'Orgnaisation',
+    description: 'Print your organisation'
+  },
+  {
+    id: 'role',
+    name: 'Role',
+    label: 'Role',
+    description: 'Print your role'
+  }
+]
+
+const selectedOptions = ref([])
+const showIntermediate = ref(false)
+const allSelected = ref(false)
+
+watch(selectedOptions, () => {
+  var showIntVal, allSelVal
+  console.log(selectedOptions.value)
+  if (selectedOptions.value.length == 0) {
+    showIntVal = false
+    allSelVal = false
+  } else if (selectedOptions.value.length == 4) {
+    showIntVal = false
+    allSelVal = true
+  } else {
+    showIntVal = true
+    allSelVal = false
+  }
+  showIntermediate.value = showIntVal
+  allSelected.value = allSelVal
+  document.getElementById('select-all').checked = allSelVal
+})
+
+const selectAll = () => {
+  if (document.getElementById('select-all').checked == true) {
```

Comment:
don't use getelementbyid. you can refer to html element via the ref attribute. 
https://vuejs.org/guide/essentials/template-refs.html

Reply:
similar for the rest